src/main/java/com/aws/greengrass/mqttbridge/clients/IoTCoreClient.java

```diff
@@ -48,14 +62,36 @@ public class IoTCoreClient implements MessageClient {
         }
     };
 
+    private final MqttClientConnectionEvents connectionCallbacks = new MqttClientConnectionEvents() {
+        @Override
+        public void onConnectionInterrupted(int errorCode) {
+        }
+
+        @Override
+        public void onConnectionResumed(boolean sessionPresent) {
+            synchronized (this)  {
```

Comment:
what _this_ is this? You're inside of a class MqttClientConnectionEvents

Reply:
makes sense now, synchronize on a lock object?

Reply:
Yes that's fine